### ISSUE TITLE
pd.Timestamp behavior/output differences caused by space character placement in the parsed string

### ISSUE THREAD

**Issue by mnmnc** [pandas-dev/pandas#17265]
#### Code Sample

```python
print(pd.Timestamp('2013, 11, 04'))
print(pd.Timestamp('2013,11, 04'))
print(pd.Timestamp('2013, 11,04'))
```
#### Problem description

Input string changes only by placing spaces differently, but that causes different results. 
Above code results to following:
```
2013-11-04 00:00:00
2013-04-01 00:00:00
2013-11-01 00:00:00
```

I checked current issues opened and did not found anything related to "Timestamp and spaces placement" within parsed string.

I'm running Python 3.6.1 distributed with Anaconda.
Pandas package version: 
pandas                    0.20.1              np112py36_0

#### Expected Output
```
2013-11-04 00:00:00
2013-11-04 00:00:00
2013-11-04 00:00:00
```
#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.1.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 78 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.1
pytest: 3.0.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 5.3.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.1
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.2.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

**Comment by gfyoung** [comment]
Interesting...that is quite strange.  Documented or not, this behavior is not intuitive and should be addressed.  Thanks for reporting this @mnmnc !

cc @jreback

**Comment by jorisvandenbossche** [comment]
I first thought this would be due to `dateutil` parser since we use that in the end if there is no clear format. But it seems they have a diffferent (even stranger to me) output for those cases:

```
In [12]: import dateutil

In [13]: dateutil.parser.parse('2013, 11, 04')
Out[13]: datetime.datetime(2013, 11, 4, 0, 0)

In [14]: dateutil.parser.parse('2013,11, 04')
Out[14]: datetime.datetime(2013, 4, 16, 0, 0)

In [15]: dateutil.parser.parse('2013, 11,04')
Out[15]: datetime.datetime(2013, 11, 16, 0, 0)
```

BTW, if you want to be sure of a consistent and correct parsing, always provide a `format`

**Comment by jreback** [comment]
This is a non-ISO 8601 input. we don't guarantee anything. I suppose it should just error.

**Comment by jreback** [comment]
this should fail https://github.com/pandas-dev/pandas/blob/master/pandas/_libs/src/datetime/np_datetime_strings.c#L395

**Comment by watercrossing** [comment]
It does fail, but then `dateutil.parser.parse` is called:
https://github.com/pandas-dev/pandas/blob/17e0b13095b23ade8e64434e88300b849bfadf01/pandas/_libs/tslibs/parsing.pyx#L105
via https://github.com/pandas-dev/pandas/blob/17e0b13095b23ade8e64434e88300b849bfadf01/pandas/_libs/tslibs/conversion.pyx#L317

Now
```
>>> from dateutil.parser import parse
>>> parse("2013,11,04")
datetime.datetime(2013, 4, 9, 0, 0)
```
I suppose there is a timeformat that is specified by '%4d,%2d,%2d`?
I guess this *try everything* approach of pandas' datetime parsing is what leads to these strange results. It's a feature?

**Comment by jreback** [comment]
we do allows a lot of separators but they must all be the same (here ,)

so this is invalid (but i suppose i could modify to accept spaces in between as well)

**Comment by jorisvandenbossche** [comment]
> It's a feature?

Yes, it is a feature that in the end we use dateutil which is very flexible in what to parse (although a feature I personally would never rely upon, as you can get very inconsistent parsing between rows). But it seems to me that `dateutil` is giving wrong output here (at least I can't see the logic), so it is maybe something that can be reported over there.
